### PR TITLE
Convert parent-join mappers to parametrized form

### PR DESCRIPTION
**Pull request by romseygeek**
This converts the three parent-join mapper implementations to parametrized
form; MetaJoinFieldMapper and ParentIdFieldMapper have no builders or
merging logic as they are always created directly by the ParentJoinFieldMapper.

Relates to #62988

### CODE REVIEW

File: modules/parent-join/src/main/java/org/elasticsearch/join/mapper/ParentJoinFieldMapper.java [hunk 1]
```diff
@@ -189,7 +160,6 @@ public String typeName() {
 
         @Override
         public IndexFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName, Supplier<SearchLookup> searchLookup) {
-            failIfNoDocValues();
```
Comment (javanna) [inline modules/parent-join/src/main/java/org/elasticsearch/join/mapper/ParentJoinFieldMapper.java:192]: can you remind me why this goes away?

File: modules/parent-join/src/main/java/org/elasticsearch/join/mapper/Relations.java [hunk 2]
```diff
@@ -53,4 +61,20 @@ public int hashCode() {
     public String toString() {
         return parent + "->" + children;
     }
+
+    public static List<Relations> parse(Object fieldNode) {
```
Comment (javanna) [inline modules/parent-join/src/main/java/org/elasticsearch/join/mapper/Relations.java:65]: does this need to be public?

File: modules/parent-join/src/main/java/org/elasticsearch/join/mapper/MetaJoinFieldMapper.java [hunk 3]
```diff
@@ -92,22 +62,17 @@ public String typeName() {
...
         }
 
         @Override
         public Object valueForDisplay(Object value) {
-            if (value == null) {
-                return null;
-            }
-            BytesRef binaryValue = (BytesRef) value;
-            return binaryValue.utf8ToString();
+            throw new UnsupportedOperationException();
```
Comment (javanna) [inline modules/parent-join/src/main/java/org/elasticsearch/join/mapper/MetaJoinFieldMapper.java:75]: I am missing why this needs to change

**Review by javanna — APPROVED** [review]
left a couple of questions, LGTM otherwise

File: modules/parent-join/src/main/java/org/elasticsearch/join/mapper/ParentJoinFieldMapper.java (hunk 1 above)
Comment (romseygeek) [reply modules/parent-join/src/main/java/org/elasticsearch/join/mapper/ParentJoinFieldMapper.java:192]: you can't disable docvalues on the field mapper, so this never fails

File: modules/parent-join/src/main/java/org/elasticsearch/join/mapper/MetaJoinFieldMapper.java (hunk 3 above)
Comment (romseygeek) [reply modules/parent-join/src/main/java/org/elasticsearch/join/mapper/MetaJoinFieldMapper.java:75]: It should never be called, as the mapper is just a placeholder for holding the Joiner.

File: modules/parent-join/src/main/java/org/elasticsearch/join/mapper/Relations.java (hunk 2 above)
Comment (romseygeek) [reply modules/parent-join/src/main/java/org/elasticsearch/join/mapper/Relations.java:65]: It does not, I'll make it package-private.

**Comment by romseygeek** [comment]
@elasticmachine run elasticsearch-ci/2

**Pull request merged**